### PR TITLE
upgrades config.vm.box to 'ubuntu/trusty64'

### PR DESCRIPTION
**Pull request by rabellamy**
The [PPA](https://launchpad.net/~ondrej/+archive/ubuntu/apache2) that is being installed for `apache2` and `ibapache2-mod-fastcgi` has been updated and the minimum version for Ubuntu is currently [14.04 LTS (Trusty Tahr)](http://releases.ubuntu.com/14.04/).

Upgrading to [14.04](https://atlas.hashicorp.com/ubuntu/boxes/trusty64) helped resolve this error:

```bash
fatal: [dev]: FAILED! => {"cache_update_time": 0, "cache_updated": false, "changed": false, "failed": true, "msg": "'/usr/bin/apt-get -y -o \"Dpkg::Options::=--force-confdef\" -o \"Dpkg::Options::=--force-confold\"   install 'apache2'' failed: E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).\n", "stderr": "E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).\n", "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nYou might want to run 'apt-get -f install' to correct these:\nThe following packages have unmet dependencies:\n libapache2-mod-fastcgi : Depends: apache2.2-common (>= 2.2.4) but it is not going to be installed\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information...", "You might want to run 'apt-get -f install' to correct these:", "The following packages have unmet dependencies:", " libapache2-mod-fastcgi : Depends: apache2.2-common (>= 2.2.4) but it is not going to be installed"]}
```
I upgraded my local version of Ansible to 2.0.2.0 and received multiple warnings during provisioning like:

```bash
[DEPRECATION WARNING]: Using bare variables is deprecated. Update your
playbooks so that the environment value uses the full variable syntax
('{{variable}}').
This feature will be removed in a future release.
Deprecation warnings can be disabled by setting deprecation_warnings=False in
ansible.cfg.
```
Updating the playbooks to use the [full variable syntax](http://docs.ansible.com/ansible/YAMLSyntax.html#gotchas) mitigated these warnings.

* updates config.vm.box to 'ubuntu/trusty64'
* updates ruby version to 2.2.5
* updates playbooks to use the full variable syntax and wraps all 'name' values in double quotes for consistency

@scottrigby, @cweagans, @ericduran and @conortm

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -70,11 +70,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   # please see the online documentation at vagrantup.com.
 
   # Every Vagrant virtual environment requires a box to build off of.
-  config.vm.box = 'pubstack'
-
-  # The url from where the 'config.vm.box' box will be fetched if it
-  # doesn't already exist on the user's system.
-  config.vm.box_url = 'http://files.vagrantup.com/precise64.box'
+  config.vm.box = 'ubuntu/trusty64'
```
Comment (scottrigby) [inline Vagrantfile]: This is a much cleaner, and future-proof way to go (for example, try to visit http://files.vagrantup.com right now LOL). I don't think this option was documented (if it was available) when we first put this together 😄

**Comment by scottrigby** [comment]
The rest looks good based on the updated Ansible version too. To me this looks 👍  but I am not in a position to test it right now (I'm step debugging on my local pubstack right now so I can't redo my local setup for a bit). I will try to do that soon, but if anyone else can test this in the meantime that would be great.

**Comment by cweagans** [comment]
I think there are some smaller PRs that could be separated from this one - for instance, just changing the box is pretty self-contained.

We should also be somewhat careful about bumping versions of things, since the goal is to be exactly where Acquia Hosting is at.

**Comment by scottrigby** [comment]
@cweagans True. The other thing is NBCU is moving everything off of Acquia LOL… but I agree we should keep version parity as long as we're supporting pubstack, since that's it's primary goal (I'll keep comments about whether or not to support it in #194 ).

**Comment by helior** [comment]
FWIW, I patched Pubstack locally with this PR and it worked out just fine. Most importantly it resolved the fatal error posted in the description. 

👍

**Comment by scottrigby** [comment]
Getting back to this, in the end we will almost definitely do #194, but as I just noted in https://github.com/NBCUTechnology/pubstack/issues/194#issuecomment-249311637 We will need to get some patternlab equivalent before we officially deprecate pubstack.

And right now there are some NBCU devs frontend who need to use this specifically for the patternlab directory inside their project repo. Because of that I want to merge this PR.

I agree with @cweagans that we could potentially split this apart, but for now this just gets our current pubstack to be able to actually install - and this may (possibly) be the last functional PR we every merge on the project. I'm no longer worried about Acquia parity / version bump though, because we're now off of Acquia hosting entirely.

**Pull request merged**